typescript/src/ethereum.ts

```diff
@@ -1099,4 +1100,80 @@ export class TBTCToken
       blockTag: blockNumber ?? "latest",
     })
   }
+
+  // eslint-disable-next-line valid-jsdoc
+  /**
+   * @see {ChainTBTCToken#requestRedemption}
+   */
+  async requestRedemption(
+    requestRedemptionData: RequestRedemptionData
+  ): Promise<Hex> {
+    const { vault, amount, ...restData } = requestRedemptionData
+
+    const extraData = this.buildRequestRedemptionData(restData)
+
+    const tx = await sendWithRetry<ContractTransaction>(async () => {
+      return await this._instance.approveAndCall(
+        vault.identifierHex,
```

Comment:
I think we should resolve the vault address instead of requiring it to be passed in parameters.

EDIT: see the comment in `redemption.ts`.

Reply:
06601b9